src/functions/sendCandidateResults/application/service/categories/C/fault-provider-cat-c.ts

```diff
@@ -0,0 +1,119 @@
+import { CatCUniqueTypes } from '@dvsa/mes-test-schema/categories/C';
+import { Fault } from '../../../../domain/fault';
+import { CompetencyOutcome } from '../../../../domain/competency-outcome';
+import {
+  convertBooleanFaultObjectToArray,
+  convertNumericFaultObjectToArray,
+  getCompletedManoeuvres,
+} from '../../fault-provider';
+import { Competencies } from '../../../../domain/competencies';
+import { QuestionOutcome, QuestionResult } from '@dvsa/mes-test-schema/categories/common';
+
+export const getDrivingFaultsCatC = (testData: CatCUniqueTypes.TestData | undefined): Fault [] => {
+  const drivingFaults: Fault[] = [];
+
+  if (!testData) {
+    throw new Error('No Test Data');
+  }
+
+  if (testData.drivingFaults) {
+    convertNumericFaultObjectToArray(testData.drivingFaults)
+      .forEach(fault => drivingFaults.push(fault));
+  }
+
+  getNonStandardFaultsCatC(testData, CompetencyOutcome.DF)
+    .forEach(fault => drivingFaults.push(fault));
+
+  return drivingFaults;
+};
+
+export const getSeriousFaultsCatC = (testData: CatCUniqueTypes.TestData | undefined): Fault [] => {
+  const seriousFaults: Fault[] = [];
+
+  if (!testData) {
+    throw new Error('No Test Data');
+  }
+
+  if (testData.seriousFaults) {
+    convertBooleanFaultObjectToArray(testData.seriousFaults)
+      .forEach(fault => seriousFaults.push(fault));
+  }
+
+  getNonStandardFaultsCatC(testData, CompetencyOutcome.S)
+    .forEach(fault => seriousFaults.push(fault));
+
+  if (getVehicleCheckFaultCount(testData.vehicleChecks as CatCUniqueTypes.VehicleChecks, CompetencyOutcome.DF) === 5) {
+    seriousFaults.push({ name: Competencies.vehicleChecks, count: 1 });
+  }
+
+  return seriousFaults;
+};
+
+const getVehicleCheckFaultCount = (
+  vehicleChecks: CatCUniqueTypes.VehicleChecks,
+  faultType: QuestionOutcome): number => {
+  let questionCount: number = 0;
+
+  if (vehicleChecks) {
```

Comment:
You can get rid of that 1 level of indentation by returning early eg.: ``If(!vechicleChecks) { return questionCount; }``